azext_iot/_factory.py

```diff
@@ -46,6 +46,24 @@ def iot_hub_service_factory(cli_ctx, *_):
     return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_IOTHUB).iot_hub_resource
 
 
+def iot_hub_full_service_factory(cli_ctx, *_):
```

Comment:
I don't currently see enough reason to make a new factory function just to return a higher scope for the client, instead the existing `iot_hub_service_factory` can be modified to return the  higher scope, and existing usage (which is limited to device identity import/export, discovery, and tests) can be modified. An extending idea is to have the factory function take a new optional parameter specific the attribute of the client to return.

Reply:
Ok, go ahead and modify the existing, but of course ensure everything works :sunglasses: